src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs

```diff
@@ -64,6 +64,12 @@ internal class ConnectionManager : IConnectionManager {
             _userConnections = new ConcurrentDictionary<Uri, IConnection>(userConnections);
 
             UpdateRecentConnections();
+
+            _shell.Idle += OnIdle;
+        }
+
+        private void OnIdle(object sender, EventArgs e) {
```

Comment:
Then idle should work since it is fired by OLE infrastructure when app is fully initialize (comes from MSO code, formerly from Office). Idle is simpler than a separate package anyway.